ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java

```diff
@@ -96,10 +87,13 @@ private void collectData(File[] replicas, ArrayList<String> replicasList, Map<St
         File[] indexFiles = replica.listFiles(PersistentIndex.INDEX_SEGMENT_FILE_FILTER);
         long keysProcessedforReplica = 0;
         Arrays.sort(indexFiles, PersistentIndex.INDEX_SEGMENT_FILE_COMPARATOR);
+        int totalIndexCount = indexFiles.length;
+        int currentIndexCount = 0;
+        logger.info("Processing index files for replica " + replica.getName());
         for (File indexFile : indexFiles) {
           keysProcessedforReplica +=
               dumpIndexTool.dumpIndex(indexFile, replica.getName(), replicasList, new ArrayList<String>(),
-                  blobIdToStatusMap, indexStats, false);
+                  blobIdToStatusMap, indexStats, false, currentIndexCount++ >= (totalIndexCount - 2));
```

Comment:
find it more readable if you use a `for(int i = 0; i < indexFiles.length; i++)` and did `i == indexFiles.length -1` here.